### ISSUE TITLE
Avoiding infinite recursion

### ISSUE THREAD

**Issue by Glitch0011** [Palindrom/JSONPatcherProxy#3]
Hi all

I am using the observer pattern to monitor an object in a game I'm writing. In order to do things such as derive absolute positions of objects from their parents it helps to store objects recursively.

One solution would be to store a weird relationship object outside of the observed object, however an easier solution would be for the proxy-observer to ignore specific objects.

My off-the-cuff plan is to fork this project and implement a predicate callback which is fired probably around jsonpatcherproxy.js #155 which'd allow the user to choose whether or not to watch an object (probably based on whether the object's name starts with a _ or something).

You guys are the experts here so I'd love your advice over whether I'm missing some glaring flaw here, or shouldn't be doing this as all. Or if you like the idea and would like me to make a PR when I'm finished.

Tom

**Comment by alshakero** [comment]
@Glitch0011 Hi!

TL;DR: add `_isProxified` property to the object you want to dismiss.

---

Well, your use case is legit for sure. We do have an ignoring mechanism in our Palindrom library which sits on top of this in our stack. So ignoring properties is a legit use case.

But we try to keep this library as fast as possible. And adding an extra `if starts with _` to every object modification that ever happens, only to cover an edge case that happens once in a while, doesn't seem like a great idea. 

I suggest two solutions:
1. [Already implemented and ready] Add a property to the objects that you don't want to observe, by adding `_isProxified` property to your object, you disable the observation. 

2. Use the library as is (without forking, to enjoy our maintenance efforts, and to share yours!), but extend it to override [this function in the prototype](https://github.com/Palindrom/JSONPatcherProxy/blob/master/src/jsonpatcherproxy.js#L55) with your own version. I think this is cleaner and smarter than forking. You can add `key.toString().startsWith('_')` to [this line](https://github.com/Palindrom/JSONPatcherProxy/blob/master/src/jsonpatcherproxy.js#L75) and you're done.


However, there is an easy solution that is definitely better than forking and maintaining on your own.

**Comment by Glitch0011** [comment]
Excellent idea with the overridden prototype function, I'll go for that.

Although, given all this was to do exactly what your Palindrom project does I might just dump my vaguely working MVC framework and use that xD

Maybe a take-away from this would be some examples in the repo? They help a lot make a project approachable.

Many thanks
We can close this.
